sonic_y_cable/y_cable.py

```diff
@@ -1097,11 +1128,135 @@ def check_if_nic_lanes_active(physical_port):
+            if y_cable_validate_read_data(readOut, 1, physical_port, "firmware info") == EEPROM_READ_DATA_INVALID:
+                return EEPROM_ERROR
+            data[byteIdx] = readOut[0]
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to get NIC lanes active")
+        return -1
+
+    result = {}
+    NUM_MCU_SIDE = 3
+    base_addr = int(target * (FIRMWARE_INFO_PAYLOAD_SIZE / NUM_MCU_SIDE))
+    rev_major_slot1 = struct.unpack_from('<B', data[0 + base_addr:  1 + base_addr])[0]
+    rev_minor_slot1 = struct.unpack_from('<B', data[2 + base_addr:  3 + base_addr])[0]
+    rev_build_lsb_slot1 = struct.unpack_from('<B', data[4 + base_addr:  5 + base_addr])[0]
+    rev_build_msb_slot1 = struct.unpack_from('<B', data[5 + base_addr:  6 + base_addr])[0]
+    rev_major_slot2 = struct.unpack_from('<B', data[1 + base_addr:  2 + base_addr])[0]
+    rev_minor_slot2 = struct.unpack_from('<B', data[3 + base_addr:  4 + base_addr])[0]
+    rev_build_lsb_slot2 = struct.unpack_from('<B', data[6 + base_addr:  7 + base_addr])[0]
+    rev_build_msb_slot2 = struct.unpack_from('<B', data[7 + base_addr:  8 + base_addr])[0]
+    slot_status = struct.unpack_from('<B', data[8 + base_addr:  9 + base_addr])[0]
+
+    if (rev_major_slot1 == 0 and rev_minor_slot1 == 0 and rev_build_lsb_slot1 == 0 and rev_build_msb_slot1 == 0 and rev_major_slot2 == 0 and rev_minor_slot2 == 0 and rev_build_lsb_slot2 == 0 and rev_build_msb_slot2 == 0):
+        return None
+    else:
+        build_slot1 = chr(rev_build_lsb_slot1) + chr(rev_build_msb_slot1)
+        version_slot1 = str(rev_major_slot1) + "." + str(rev_minor_slot1)
+        build_slot2 = chr(rev_build_lsb_slot2) + chr(rev_build_msb_slot2)
+        version_slot2 = str(rev_major_slot2) + "." + str(rev_minor_slot2)
+        if slot_status & 0x01:
+            run_slot1 = True
+        else:
+            run_slot1 = False
+        if slot_status & 0x02:
+            commit_slot1 = True
+        else:
+            commit_slot1 = False
+        if slot_status & 0x04:
+            empty_slot1 = True
+        else:
+            empty_slot1 = False
+        if slot_status & 0x10:
+            run_slot2 = True
+        else:
+            run_slot2 = False
+        if slot_status & 0x20:
+            commit_slot2 = True
+        else:
+            commit_slot2 = False
+        if slot_status & 0x40:
+            empty_slot2 = True
+        else:
+            empty_slot2 = False
+
+        result["build_slot1"] = build_slot1
+        result["version_slot1"] = version_slot1
+        result["build_slot2"] = build_slot2
+        result["version_slot2"] = version_slot2
+        result["slot_status"] = slot_status
+        result["run_slot1"] = run_slot1
+        result["run_slot2"] = run_slot2
+        result["commit_slot1"] = commit_slot1
+        result["commit_slot2"] = commit_slot2
+        result["empty_slot1"] = empty_slot1
+        result["empty_slot2"] = empty_slot2
```

Comment:
Suggest reducing number of lines here:

```python
        result["build_slot1"] = chr(rev_build_lsb_slot1) + chr(rev_build_msb_slot1)
        result["version_slot1"] = str(rev_major_slot1) + "." + str(rev_minor_slot1)
        result["build_slot2"] = chr(rev_build_lsb_slot2) + chr(rev_build_msb_slot2)
        result["version_slot2"] = str(rev_major_slot2) + "." + str(rev_minor_slot2)
        result["run_slot1"] = True if slot_status & 0x01 else False
        result["commit_slot1"] = True if slot_status & 0x02 else False
        ...
```

Reply:
fixed

Reply:
Thanks. I see you retained the intermediate variables (e.g., `build_slot1`). They only get assigned and then are assigned to `result["xxxx"]`, so they seem unnecessary and just double the lines of code needed. I suggest following my example above and assign the values directly into the respective `result` element.

Reply:
fixed